larpandoradlcontent/LArVertex/DlVertexingBaseAlgorithm.cc

```diff
@@ -133,11 +133,50 @@ void DlVertexingBaseAlgorithm::GetHitRegion(const CaloHitList &caloHitList, floa
         const float zAsymmetry{nHitsUpstream / static_cast<float>(nHitsViewTotal)};
 
         const float xSpan{m_driftStep * (m_width - 1)};
-        xMin = xVtx - xAsymmetry * xSpan;
-        xMax = xMin + (m_driftStep * (m_width - 1));
+        if (xAsymmetry <= 0.5f)
+        {
+            xMin = xVtx - xAsymmetry * xSpan;
+            xMax = xMin + xSpan;
+        }
+        else
+        {
+            xMax = xVtx + (1 - xAsymmetry) * xSpan;
+            xMin = xMax - xSpan;
+        }
         const float zSpan{pitch * (m_height - 1)};
-        zMin = zVtx - zAsymmetry * zSpan;
-        zMax = zMin + zSpan;
+        if (zAsymmetry <= 0.5f)
+        {
+            zMin = zVtx - zAsymmetry * zSpan;
+            zMax = zMin + zSpan;
+        }
+        else
+        {
+            zMax = zVtx + (1 - zAsymmetry) * zSpan;
+            zMin = zMax - zSpan;
+        }
+
+        // Avoid the estimated vertex being too close to edges (resolution might push the true vertex outside of the frame)
+        const float tol{5.f};
```

Comment:
There are some hard-coded 5.f and 10.f's in here.  Do they depend on detector granularity and/or size?

Reply:
Hi Tom, detector granularity isn't relevant here, though detector size is - here the tolerance was chosen pragmatically to cover the worst case scenario in a DUNE-FD. There are alternatives here where resolution could be calculated dynamically, or set according to the worst case for a given detector.